hdr10plus_tool: 1.7.0 -> 1.7.1

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for hdr10plus_tool is: CLI utility to work with HDR10+ in HEVC files.

meta.homepage for hdr10plus_tool is: https://github.com/quietvoid/hdr10plus_tool

meta.changelog for hdr10plus_tool is: https://github.com/quietvoid/hdr10plus_tool/releases/tag/1.7.1


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 1.7.1 with grep in /nix/store/ig9na61ig1g0yknx0hhqxmavj1rpp4jx-hdr10plus_tool-1.7.1
- found 1.7.1 in filename of file in /nix/store/ig9na61ig1g0yknx0hhqxmavj1rpp4jx-hdr10plus_tool-1.7.1

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
4 total rebuild path(s)

4 package rebuild(s)

First fifty rebuilds by attrpath

hdr10plus
hdr10plus_tool
svt-av1-psy
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/ig9na61ig1g0yknx0hhqxmavj1rpp4jx-hdr10plus_tool-1.7.1 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A hdr10plus_tool https://github.com/r-ryantm/nixpkgs/archive/2c1024f3ae1a36d8b15d39acd610a38e9f9d95e6.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/2c1024f3ae1a36d8b15d39acd610a38e9f9d95e6#hdr10plus_tool
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/ig9na61ig1g0yknx0hhqxmavj1rpp4jx-hdr10plus_tool-1.7.1
ls -la /nix/store/ig9na61ig1g0yknx0hhqxmavj1rpp4jx-hdr10plus_tool-1.7.1/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>hdr10plus</li>
    <li>hdr10plus.dev</li>
    <li>svt-av1-psy</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hdr10plus_tool</li>
  </ul>
</details>

---

###### Maintainer pings

cc @johnrtitor for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
hdr10plus_tool: 1.7.0 -> 1.7.1

illdefined commented:
> Humm hdr10plus failed. Need to investigate.

I am not quite sure where to start with this, but I’ll play around with it on the weekend:

```
Error: CliError { error: Some(failed to parse manifest at `/build/source/Cargo.toml`                                                                                         
                                                                                                                                                                             
Caused by:                                                                                                                                                                   
    feature `edition2024` is required                                                                                                                                        
                                                                                                                                                                             
    The package requires the Cargo feature called `edition2024`, but that feature is not stabilized in this version of Cargo (1.79.0).                                       
    Consider adding `cargo-features = ["edition2024"]` to the top of Cargo.toml (above the [package] table) to tell Cargo you are opting in to use this unstable feature.    
    See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#edition-2024 for more information about the status of this feature.                                  
    ), exit_code: 101 }                                                                                                                                                      
```

JohnRTitor commented:
CC @quietvoid

JohnRTitor commented:
Note that `nixos-unstable` channel has cargo 1.88.0 at the moment.

quietvoid commented:
> The minimum Rust version to build `hdr10plus_tool` is 1.85.0.

JohnRTitor commented:
> > The minimum Rust version to build `hdr10plus_tool` is 1.85.0.

as I said nixpkgs has cargo 1.88.0 at the moment.

```
Running phase: buildPhase
Error: CliError { error: Some(failed to parse manifest at `/tmp/tmp.YYJj6b1GR2/source/Cargo.toml`

Caused by:
    feature `edition2024` is required
    
    The package requires the Cargo feature called `edition2024`, but that feature is not stabilized in this version of Cargo (1.79.0).
    Consider adding `cargo-features = ["edition2024"]` to the top of Cargo.toml (above the [package] table) to tell Cargo you are opting in to use this unstable feature.
    See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#edition-2024 for more information about the status of this feature.
    ), exit_code: 101 }
Running phase: checkPhase
Error: CliError { error: Some(failed to parse manifest at `/tmp/tmp.YYJj6b1GR2/source/Cargo.toml`

Caused by:
    feature `edition2024` is required
    
    The package requires the Cargo feature called `edition2024`, but that feature is not stabilized in this version of Cargo (1.79.0).
    Consider adding `cargo-features = ["edition2024"]` to the top of Cargo.toml (above the [package] table) to tell Cargo you are opting in to use this unstable feature.
    See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#edition-2024 for more information about the status of this feature.
    ), exit_code: 101 } 

❯  ls
assets  Cargo.lock  Cargo.toml  hdr10plus  LICENSE  README.md  src  tests

❯  cargo --version
cargo 1.88.0 (873a06493 2025-05-10)
```

JohnRTitor commented:
Okay looks like this "cargo" is actually `cargo-c`, which has been on 10.2 for a long time. https://github.com/NixOS/nixpkgs/pull/413726

10.14 release ["adds" support](https://github.com/lu-zero/cargo-c/releases/tag/v0.10.14) for cargo 1.89 hence the confusion.

JohnRTitor committed:
hdr10plus: decouple the src from hdr10plus_tool

The main reason for this is cargo-c latest version is not available in master yet.
This will need to be reverted in staging once https://github.com/NixOS/nixpkgs/pull/413726 is merged

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

JohnRTitor commented:
For now I pinned hdr10plus (the library) to 1.7.0 version of the repo.

This needs to be reverted in staging.
